### PR TITLE
Slack refactor

### PR DESCRIPTION
**Pull request by ryanpbrewster**
@leogrim @andrewconner thoughts on this?

I split out two classes from SlackController (which now only handles API endpoints).

SlackOAuthController is a redirecting controller. It handles browser requests on the way back from Slack. It also has two other calls in there right now, and I'm not sure if they really belong.

SlackAuthenicationCommander handles `processAuthorizedAction` and returns `SlackResponse`.

### CODE REVIEW

**Comment by andrewconner** [comment]
I'm good with smaller classes, Léo may have a more informed position in this case than I do.

File: kifi-backend/modules/shoebox/conf/shoeboxService.routes [hunk 1]
```diff
@@ -13,7 +13,11 @@ GET     /contacts/import/failure    @com.keepit.controllers.website.ContactsImpo
 GET     /contacts/refresh           @com.keepit.controllers.core.OAuth2Controller.refreshContacts(abookId:ExternalId[ABookInfo], provider:Option[String])
 GET     /oauth2/start/:provider     @com.keepit.controllers.core.OAuth2Controller.start(provider:String, stateToken:Option[String], approvalPromptOpt:Option[String])
 GET     /oauth2/callback/:provider  @com.keepit.controllers.core.OAuth2Controller.callback(provider:String)
-GET     /oauth2/slack               @com.keepit.controllers.website.SlackController.registerSlackAuthorization(code: Option[String] ?= None, state: String)
+GET     /oauth2/slack               @com.keepit.controllers.website.SlackOAuthController.registerSlackAuthorization(code: Option[String] ?= None, state: String)
+
+# These really need to be non-site endpoints
+GET     /site/slack/add                      @com.keepit.controllers.website.SlackOAuthController.addSlackTeam(slackTeamId: Option[SlackTeamId] ?= None)
+GET     /site/organizations/create/slack     @com.keepit.controllers.website.SlackOAuthController.createSlackTeam(slackTeamId: Option[SlackTeamId] ?= None)
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/conf/shoeboxService.routes:20]: @leogrim can these move to `site.routes` and live in the `SlackAuthRouter`?

File: kifi-backend/modules/shoebox/conf/shoeboxService.routes [hunk 2]
```diff
@@ -467,12 +471,9 @@ GET     /site/rpb/user/:userId/libraries   @com.keepit.controllers.website.UserC
 GET     /site/rpb/org/:orgId/libraries     @com.keepit.controllers.website.UserController.rpbGetOrgLibraries(orgId: PublicId[Organization], fromId: Option[String] ?= None, offset: Int ?= 0, limit: Int ?= 10)
 
 POST  /site/libraries/:id/slack/add         @com.keepit.controllers.website.SlackController.setupLibraryIntegrations(id: PublicId[Library])
-POST  /site/libraries/:id/slack/push        @com.keepit.controllers.website.SlackController.turnOnLibraryPush(id: PublicId[Library], integrationId: String)
-POST  /site/libraries/:id/slack/ingest      @com.keepit.controllers.website.SlackController.turnOnChannelIngestion(id: PublicId[Library], integrationId: String)
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/conf/shoeboxService.routes:471]: @andrewconner I don't think these are used on the frontend. Can you confirm?

Comment (andrewconner) [inline kifi-backend/modules/shoebox/conf/shoeboxService.routes:471]: Not directly, but I do know that the backend used to give these links to the frontend to call.

File: kifi-backend/modules/shoebox/conf/shoeboxService.routes (hunk 1 above)
Comment (andrewconner) [inline kifi-backend/modules/shoebox/conf/shoeboxService.routes:20]: If you do that, change the routing too. But before you move them, let me know so I can update the frontend.

Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/conf/shoeboxService.routes:20]: Probably a separate PR, but I'd like to get that done ASAP so we can move on without too much debt.

File: kifi-backend/modules/shoebox/conf/shoeboxService.routes (hunk 2 above)
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/conf/shoeboxService.routes:471]: Yeah, IMO the way we do it now is cleaner, but I'll chat with @leogrim about it again today.

**Comment by leogrim** [comment]
:+1: thanks!

**Pull request merged**